backend/src/main/java/com/example/backend/service/MovieService.java

```diff
@@ -0,0 +1,46 @@
+package com.example.backend.service;
+
+
+import com.example.backend.model.Movie;
+import com.example.backend.model.MovieDTO;
+import com.example.backend.repo.MovieDB;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+
+@Service
+public class MovieService {
+
+    private MovieDB movieRepository;
+    private IdService idService;
+
+    @Autowired
+    public MovieService(MovieDB movieRepository, IdService idService){
+        this.movieRepository = movieRepository;
+        this.idService = idService;
+    }
+
+
+    public List<Movie> getAllMovies(){
+        return movieRepository.getAllMovies();
+    }
+
+    public Movie addMovie(MovieDTO movie) {
+        Movie newMovie = Movie.builder()
+                .filmtitel(movie.getTitel())
+                .year(movie.getYear())
+                .url(movie.getUrl())
+                .box(movie.isBox())
+                .id(movie.getId())
```

Comment:
Hier würdest du die Id nicht aus dem Dto rausholen (das Dto sollte ja keine Id haben).
Sondern hier kannst du den IdService nutzen, um die Id zu generieren.

```suggestion
                .id(idService.generateID())
```